src/components/sections/Searched.jsx

```diff
@@ -0,0 +1,47 @@
+import { Link, useParams } from "react-router-dom";
+import { useSearchArticleQuery } from "../../features/newsSlice";
+import ArticleCard from "../ui/ArticleCard";
+import Wrapper from "../wrappers/Wrapper";
+
+const Searched = () => {
+    const { keyword } = useParams();
+    const { data, isSuccess } = useSearchArticleQuery(keyword);
+    return (
+        <Wrapper>
+            {isSuccess ? (
+                <div>
+                    {data.length > 0 ? (
+                        <div>
+                            <button className="bg-red-700 px-2 py-3 m-4 text-white font-semibold hover:bg-gray-600">
+                                <Link to="/">Back Home</Link>
+                            </button>
+                            <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-5 gap-3">
+                                {data.map(({ title, urlToImage, url }, index) => (
+                                    <ArticleCard
+                                        title={title}
+                                        urlToImage={urlToImage}
+                                        url={url}
+                                        key={index}
+                                    />
+                                ))}
+                            </div>
+                        </div>
+                    ) : (
+                        <div className="text-red-700 text-4xl font-bold text-center flex justify-center items-center h-screen w-full flex-col gap-5">
+                            <h1>{`Article with keyword ${keyword} was not found!!!!`}</h1>
+                            <button className="bg-red-700 px-2 py-4 m-4 text-white font-semibold hover:bg-gray-600  text-sm">
+                                <Link to="/">Go Back to Home Page</Link>
+                            </button>
+                        </div>
+                    )}
+                </div>
+            ) : (
```

Comment:
Sometimes the request can fail, and I think you want to show the loading component even when the query failed, instead, the same hook provides also an isLoading property that you can use in this case here.

Reply:
here I'm using the property **isSuccess** to check if the request has failed or succeeded. the loading Component is shown until the request succeeds and data are fetched. are you recommending to use isLoading instead?